scene/resources/dynamic_font.cpp

```diff
@@ -326,7 +326,7 @@ float DynamicFontAtSize::draw_char(RID p_canvas_item, const Point2 &p_pos, CharT
 			cpos.y -= font->get_ascent();
 			cpos.y += ch->v_align;
 			Color modulate = p_modulate;
-			if (FT_HAS_COLOR(face)) {
+			if (FT_HAS_COLOR(font->face)) {
```

Comment:
Should the other calls to `FT_HAS_COLOR(face)` get the same treatment?
```
scene/resources/dynamic_font.cpp
169:    if (FT_HAS_COLOR(face) && face->num_fixed_sizes > 0) {
309:            int error = FT_Load_Char(face, p_char, FT_HAS_COLOR(face) ? FT_LOAD_COLOR : FT_LOAD_DEFAULT);
329:                    if (FT_HAS_COLOR(face)) {
598:    int error = FT_Load_Char(face, p_char, FT_HAS_COLOR(face) ? FT_LOAD_COLOR : FT_LOAD_DEFAULT | (font->force_autohinter ? FT_LOAD_FORCE_AUTOHINT : 0) | ft_hinting);
```

Reply:
No, `draw_char` and `get_char_size` are the only methods that are always called for base data instance and then accessing fallbacks. The rest are called for the fallback data directly.

Reply:
This one is in `draw_char` though, used for the outline:
```
309:            int error = FT_Load_Char(face, p_char, FT_HAS_COLOR(face) ? FT_LOAD_COLOR : FT_LOAD_DEFAULT);
```

Reply:
This is with this PR when using emoji font as main font:
![Screenshot_20201209_104018](https://user-images.githubusercontent.com/4701338/101612489-fc2fb100-3a0a-11eb-96a0-97e1c79b9cbe.png)

Reply:
This additional diff doesn't solve it though, might be more complex than that.
```diff
diff --git a/scene/resources/dynamic_font.cpp b/scene/resources/dynamic_font.cpp
index 7027660633..ac6798cc23 100644
--- a/scene/resources/dynamic_font.cpp
+++ b/scene/resources/dynamic_font.cpp
@@ -305,10 +305,10 @@ float DynamicFontAtSize::draw_char(RID p_canvas_item, const Point2 &p_pos, CharT
 
        // use normal character size if there's no outline character
        if (p_outline && !ch->found) {
-               FT_GlyphSlot slot = face->glyph;
-               int error = FT_Load_Char(face, p_char, FT_HAS_COLOR(face) ? FT_LOAD_COLOR : FT_LOAD_DEFAULT);
+               FT_GlyphSlot slot = font->face->glyph;
+               int error = FT_Load_Char(font->face, p_char, FT_HAS_COLOR(font->face) ? FT_LOAD_COLOR : FT_LOAD_DEFAULT);
                if (!error) {
-                       error = FT_Render_Glyph(face->glyph, FT_RENDER_MODE_NORMAL);
+                       error = FT_Render_Glyph(font->face->glyph, FT_RENDER_MODE_NORMAL);
                        if (!error) {
                                Character character = Character::not_found();
                                character = const_cast<DynamicFontAtSize *>(this)->_bitmap_to_character(slot->bitmap, slot->bitmap_top, slot->bitmap_left, slot->advance.x / 64.0);
```

Reply:
I think it's how it should work, outlines do no work with bitmaps, and the aforementioned code is only for skipping bitmap character width when drawing an outline.

Reply:
Sounds good, thanks for clarifying.

Reply:
For the reference there's no outline for bitmaps emoji in master either (and embedded SVG emoji is not yet supported by freetype).

Reply:
And we have this issue to keep track of it: https://github.com/godotengine/godot/issues/34870